app/components/Checkbox/index.js

```diff
@@ -46,26 +46,23 @@ const Checkbox = ({
 
     {/* This is the control box with a check icon as children */}
     <ControlBox
-      borderWidth="1.5px" // TODO: should come from theme
-      size="14px" // TODO: should come from theme
-      borderColor={
-        isDisabled ? 'checkbox.disabled.borderColor' : 'checkbox.borderColor' // TODO: Based on theme.borderColor
-      }
+      size={0.875}
+      borderWidth={1.5}
+      borderColor="borderColor.checkbox.default"
       rounded="md"
-      _checked={
-        isDisabled
-          ? {}
-          : { bg: 'brand.500', color: 'white', borderColor: 'brand.500' } // TODO: Theme.disabled should have the styles
-      }
+      _checked={{
+        bg: 'bg.checkbox.checked',
+        color: 'white',
+        borderColor: 'borderColor.checkbox.checked',
+      }}
+      _disabled={{ borderColor: 'borderColor.checkbox.disabled' }}
+      _checkedAndDisabled={{}}
       _focus={{ borderColor: 'outline' }}
-      _hover={
-        isDisabled
-          ? {}
-          : {
-              borderColor: 'brand.500', // Theme.border
-              shadow: '0 0 0 2px #31a7d840, inset 0 0 0 2px #31a7d840',
-            }
-      }
+      _hover={{
+        borderColor: 'borderColor.checkbox.hover',
+        shadow: 'checkbox.hover',
+      }}
+      _checkedAndHover={{}}
       {...getAllowedProps(override.controlBox)}
```

Comment:
Govind is working on filtering props right ?

Reply:
yes

Reply:
Yeah, he is working on filtering the props, these props are for our component styling.